app/importers/billing_contracts_import.rb

```diff
@@ -0,0 +1,53 @@
+class BillingContractsImport
+  include ActiveModel::Model
+  require 'roo'
+
+  attr_accessor :file
+
+  def initialize(attributes = {})
+    attributes.each { |name, value| send("#{name}=", value) }
+  end
+
+  def save
+    if imported_billing_contracts.map(&:valid?).all?
+      imported_billing_contracts.each(&:save!)
+      true
+    else
+      imported_billing_contracts.each_with_index do |billing_contract, index|
+        billing_contract.errors.full_messages.each do |msg|
+          errors.add :base, "Row #{index + 3}: #{msg}"
+        end
+      end
+      false
+    end
+  rescue StandardError => e
+    errors.add :base, e.message
+    false
+  end
+
+  private
+
+  def open_spreadsheet
+    case File.extname(file.original_filename)
+    when '.csv' then Roo::CSV.new(file.path)
+    when '.xls' then Roo::Excel.new(file.path)
+    when '.xlsx' then Roo::Excelx.new(file.path)
+    else raise "Unknown file type: #{file.original_filename}"
+    end
+  end
+
+  def load_imported_billing_contracts
+    spreadsheet = open_spreadsheet
+    header = spreadsheet.row(2)
+    (3..spreadsheet.last_row).map do |i|
+      row = Hash[[header, spreadsheet.row(i)].transpose]
+      billing_contract = BillingContract.find_by(contract_num: row["contract_num"]) || BillingContract.new
```

Comment:
this part was discussed as well there will be no update with import for now
don't make functionality that it not is scope of the story